### PR TITLE
Add button expanded class for each breakpoint

### PR DESCRIPTION
**Pull request by DaSchTour**
Follow up to #7575 closing #7570

### CODE REVIEW

**Comment by IamManchanda** [comment]
@DaSchTour  as I said before

```
$responsive-expanded-button: true; // maybe `false` by default
```

Will also poke @kball

**Comment by DaSchTour** [comment]
@IamManchanda well, good we talked about this. I checked the generated CSS and saw that my usecase (expanded buttons on small viewports) wasn't covered. So I improved the and added -up and -down modifiers.

**Comment by IamManchanda** [comment]
@kball What do you say ... True or false?

File: scss/components/_button.scss [hunk 1]
```diff
@@ -271,6 +275,29 @@ $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !defau
...
+        }
+        @if $size != $-zf-zero-breakpoint {
+          @include breakpoint(#{$size} down) {
+            &.expanded-for-#{$size}-down {
+              @include button-expand;
+            }
+          }
+
+          @include breakpoint(#{$size}) {
+            &.expanded-fpr-#{$size}-up {
```
Comment (IamManchanda) [inline scss/components/_button.scss]: fpr

**Review by IamManchanda — CHANGES_REQUESTED** [review]

File: scss/components/_button.scss [hunk 2]
```diff
@@ -80,6 +80,10 @@ $button-hollow-hover-lightness: -50% !default;
 /// @type List
 $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !default;
 
+/// Additional responsive classes for .expanded
+/// @type Boolean
+$responsive-expanded-button: true !default;
```
Comment (IamManchanda) [inline scss/components/_button.scss]: @kball true or false

File: docs/pages/button.md [hunk 3]
```diff
@@ -59,6 +59,7 @@ Additional classes can be added to your button to change its size and shape.
 <a class="button large" href="#">So Large</a>
 <a class="button expanded" href="#">Such Expand</a>
 <a class="button small expanded" href="#">Wow, Small Expand</a>
+<a class="button small expanded-for-large" href="#">Wow, Small + Expand for Large</a>
```
Comment (rafibomb) [inline docs/pages/button.md]: In terms of class naming - let's match our current naming scheme for the grid. 
large-expanded, medium-expanded etc https://github.com/zurb/foundation-sites/blob/develop/scss/grid/_flex-grid.scss#L216

File: scss/components/_button.scss (hunk 2 above)
Comment (rafibomb) [reply scss/components/_button.scss]: I say false

Also - let's rename the variable so it matches the others
$button-responsive-expanded

File: docs/pages/button.md (hunk 3 above)
Comment (IamManchanda) [reply docs/pages/button.md]: Yup `large-expanded`

**Comment by rafibomb** [comment]
After these 2 changes - looks good to merge! @DaSchTour are you able to make the updates so we can merge today?

**Comment by DaSchTour** [comment]
@rafibomb done 😄

**Comment by rafibomb** [comment]
One question came up while testing this. If we set `$button-responsive-expanded: false !default;` to false then how do CSS download users get this feature?

**Comment by rafibomb** [comment]
In the meantime I updated the docs for this.

**Comment by DaSchTour** [comment]
I guess CSS download users are and will be more and more limited in the number of features the get. In the long term I guess there will be more such "hidden" features. The only way so solve this would be to create CSS for every component with different settings. This way it would also be possible to only include certain components, just like in JavaScript.

**Pull request merged**